File: ios/Jazzify/Defense/Engine/DefenseChordHudLabels.swift
import Foundation

struct DefenseChordHudLabels: Equatable {
    let current: String

    static func make(chordNames: [String], chordIndex: Int) -> DefenseChordHudLabels {
        guard !chordNames.isEmpty else {
            return DefenseChordHudLabels(current: "-")
        }
        let safeIndex = min(max(chordIndex, 0), chordNames.count - 1)
        let current = chordNames[safeIndex]
        return DefenseChordHudLabels(current: current)
    }
}

File: src/game/defense/defenseChordHudLabels.ts
export interface DefenseChordHudLabels {
  readonly current: string;
}

export const getDefenseChordHudLabels = (
  chordNames: readonly string[],
  chordIndex: number,
): DefenseChordHudLabels => {
  if (chordNames.length === 0) {
    return { current: '-' };
  }

  const safeIndex = Math.min(Math.max(chordIndex, 0), chordNames.length - 1);
  const current = chordNames[safeIndex] ?? '-';
  return { current };
};
